src/Cardano/Shell/Presets.hs

```diff
@@ -20,25 +22,25 @@ import           Cardano.Shell.Constants.Types (Block (..),
 -- Cardano Mainnet Configuration
 --------------------------------------------------------------------------------
 
-mainnetConfiguration :: CardanoConfiguration
+mainnetConfiguration :: PartialCardanoConfiguration
 mainnetConfiguration =
-  CardanoConfiguration
-    { ccLogPath             = "./logs/"
-    , ccLogConfig           = "./configuration/log-configuration.yaml"
-    , ccDBPath              = "./db/"
-    , ccApplicationLockFile = ""
-    , ccCore =
-        pure Core
-          { coGenesis =
-              pure Genesis
-                { geSrc             = "mainnet-genesis.json"
-                , geGenesisHash     = "89d9b5a5b8ddc8d7e5a6795e9774d97faf1efea59b2caf7eaf9f8c5b32059df4"
-                , gePrevBlockHash   = "5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb"
+  PartialCardanoConfiguration
```

Comment:
More weirdness. 
What is the point of separation of `CardanoConfiguration` and `PartialCardanoConfiguration` if there is no point to it :smile:

Reply:
I guess what is missing here is the picture of how we arrive at the final value of `CardanoConfiguration`, in a generic way.  This goes back to the call we had between the three of us with @CodiePP.

I do agree that in the _current_ form, there's no use for the specific `PartialCardanoConfiguration`, since we're not merging them yet.  But this gives us an opportunity for transparent merging of feature-provided `PartialCardanoConfiguration`'s.

So the short answer is -- it's there to provide mergeability.